src/v2/view-builder/utils/AuthenticatorUtil.js

```diff
@@ -297,8 +310,8 @@ export function getAuthenticatorDataForVerification(authenticator) {
   return getAuthenticatorData(authenticator, true);
 }
 
-export function getIconClassNameForBeacon(authenticatorKey) {
-  return getAuthenticatorData({ authenticatorKey }).iconClassName;
+export function getIconClassNameForBeacon(authenticatorKey, idvName) {
+  return getAuthenticatorData({ authenticatorKey }).iconClassName || getIDProofingData(idvName)?.iconClassName;
```

Comment:
The way the `getIDProofingData` method is written, it will always return an object so you don't need the optional chain. 

Hard to tell in JS (vs TS), the `idvName` is an optional second param that shouldn't break existing calls to `getIconClassNameForBeacon`? It'll fall through all cases in the switch if I'm reading correctly.

So it'd be `getIDProofingData(idvName?: string)` and `getIconClassNameForBeacon(authenticatorKey: string, idvName?: string)` in TS

```suggestion
  return getAuthenticatorData({ authenticatorKey }).iconClassName || getIDProofingData(idvName).iconClassName;
```

Reply:
`getIconClassNameForBeacon` has been called in only one [place](https://github.com/okta/okta-signin-widget/blob/ea0bcf2ce326f0fef2785074138722da9e840099/src/v2/view-builder/components/BaseAuthenticatorView.js#L25) and i am passing both values. IN real scenario , we will either have `authenticatorKey` or `IdvName`.

Yes you are right, we may not need optional chaining (?:) but consider this scenario where we introduced new IDV `clear` shortly in backend and in the UI if we do not have the supported code yet, then we will get `undefined` ( `{}.iconClassName` ) instead of `{}`. Anyway this scenario will be handled gracefully in [template](https://github.com/okta/okta-signin-widget/blob/ea0bcf2ce326f0fef2785074138722da9e840099/src/v2/view-builder/components/HeaderBeacon.js#L26).

So i feel it should not make a much impact. I prefer to keep an empty object instead of `undefined`. If you have a strong opinion i can change it.

Reply:
`getIconClassNameForBeacon` has been called in only one [place](https://github.com/okta/okta-signin-widget/blob/ea0bcf2ce326f0fef2785074138722da9e840099/src/v2/view-builder/components/BaseAuthenticatorView.js#L25) and i am passing both values. In real scenario , we will either have authenticatorKey or IdvName.
Yes you are right, we may not need optional chaining (?.)